### ISSUE TITLE
Notify decoupled site when content is published

### ISSUE THREAD

**Issue by Snugug** [Lullabot/ibm-dte#24]
When a piece of content or a taxonomy term is published for the first time, when a translation is published, or when existing content or taxonomy term is updated and a new revision is published, we need to be able to notify our decoupled site of the change. Allowing this allows our decoupled front-ends to cache their responses and only update the cache on content change, reducing hits to API origin.

---

Original thought on this is firing a webhook, but I'm open to other options.

**Comment by dawehner** [comment]
Hi 👋 

I see a couple of possibilities:

* As you said, webhooks. They are indeed a nice solution, with a lot of flexibility for the clients. I'm not sure you use the [consumers](http://drupal.org/project/consumers) module yet, but it provides a nice abstraction to do something per consumer. In this case we could provide different webhook URLs per consumer
* Pubsubhub (now called websub apparently) is a layer on top of webhooks. It introduced the concept of a hub to help with subscribing and distributing the changes
* Depending on the usecases I could also see you using a more traditional queue system with all of your consumers subscribing to it. On the other hand I'm not sure this would work well with mobile phone apps, or offline ready single page apps.

TL;DR
I think webhooks would be a good starting point. Depending on the usecases pubsubhub might give you more automated subscribing mechanisms.

**Comment by Snugug** [comment]
Thanks for the feedback! A little more context about our architecture; our decoupled Drupal instance is doesn't itself have multiple front-ends connected to it. Instead, it has a single service that communicates with it; an API normalization layer or API Gateway. That API Gateway does things like cache API requests, control staging/production access, compiles multiple services we provide under a single unified API layer, etc…

So in the case above, our "decoupled site" is _actually_ our API Gateway, which feeds the various front-ends consuming our content.

**Comment by sethlbrown** [comment]
Notes from Friday, September 28, 2018
----------------
Webhooks, allows configuration by events for what fires webhook
Sam thinks this module will work
Use case: Have 2.5mm page views with Pantheon, Pantheon charges based on anything that requires PHP to bootstrap, don't want to hit origin API whenever someone updates a page, want to notify a caching service that content has been updated or published, update hook will likely work, but Sam needs to do additional research

Next steps: Sam to investigate.

**Comment by sethlbrown** [comment]
Webhook module will work for us.
